website/server/libs/user/validation.js

```diff
@@ -1,13 +1,20 @@
 import bannedSlurs from '../bannedSlurs';
-import { getMatchesByWordArray } from '../stringUtils';
+import bannedWords from '../bannedWords';
+import {
+  getMatchesByWordArray,
+  normalizeUnicodeString,
+  removePunctuationFromString,
+} from '../stringUtils';
 import forbiddenUsernames from '../forbiddenUsernames';
 
-const bannedSlurRegexs = bannedSlurs.map(word => new RegExp(`.*${word}.*`, 'i'));
+export function stringContainsProfanity (str, profanityType = 'bannedWord') {
```

Comment:
I'm a bit worried about constructing the regexes here every time the function is invoked instead of doing it outside of the function like it was done before

Reply:
@paglias Since we need two possible arrays of regexes - one for slurs and one for profanity - wouldn't moving it outside mean that we always construct one of the regex sets for no reason? Since the function only ever wants one. I have moved outside for now anyway and it might be more readable this way.